src/main/java/htsjdk/samtools/DuplicateScoringStrategy.java

```diff
@@ -80,6 +86,8 @@ public static short computeDuplicateScore(final SAMRecord record, final ScoringS
                         score += SAMUtils.getMateCigar(record).getReferenceLength();
                     }
                     break;
+                case RANDOM:
+                    score += (short) (hasher.hashUnencodedChars(record.getReadName()) >> 16);
```

Comment:
I think that there should be no difference in the randomness, due to how
murmur works (lots of >>>'s, which mixes up low and high order bits), so
I'm happy.

On Fri, Aug 19, 2016 at 2:54 PM, Tim Fennell <notifications@github.com>
wrote:

> In src/main/java/htsjdk/samtools/DuplicateScoringStrategy.java
> <https://github.com/samtools/htsjdk/pull/688#discussion_r75534218>:
>
> > @@ -80,6 +86,8 @@ public static short computeDuplicateScore(final SAMRecord record, final ScoringS
> >                          score += SAMUtils.getMateCigar(record).getReferenceLength();
> >                      }
> >                      break;
> > +                case RANDOM:
> > +                    score += (short) (hasher.hashUnencodedChars(record.getReadName()) >> 16);
>
> No particular reason other than that >> 16 came to mind for me first.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/samtools/htsjdk/pull/688/files/1a5616268a3d8f0a69a734a7ab26a7ca3ef4f0b2#r75534218>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACnk0lD5n5D0lZ7vnJ2e8oNbksmwEBuhks5qhfvWgaJpZM4Joo0U>
> .
>